src/handlers/authenticationHandlers.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+import { pathObj } from './../utils/generalVariables';
+
+export const loginHandler = ({ username, password }) => {
+  // With the finalization of the database schema, more checks can be implemented (with separate error-messages)
+  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 8 ) {
+    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  }
+
+  return axios.post(`${pathObj.loginPath}`, { 
+    username, 
+    password,
+  })
+    .then(res => {
+      if (res.token) {
+        checkAndStoreToken(res.data.token);
+        return res.data.user
+      } else {
+        return new Error('Oh no, there was no token returned by the database!')
+      }
+    })
+    .catch(error => {
+      return new Error(error)
+    })
+}
+
+// There is no `addUserHandler` since the only avenue for adding new users should be when an admin registers a new user
+// Therefore, whenever a new user is added, this method should be used. In essence, it replicates the CRUD functionality.
+export const registrationHandler = ({ username, password }) => {
+  // Once database schema is finalized, this conditional check could be refactored into a separate utility function.
+  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 8 ) {
+    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  }
+  
+  return axios.post(`${pathObj.registrationPath}`, { 
+    username, 
+    password,
+  })
```

Comment:
You will need to send the user token with the request to validate admin status in the backend handling

Reply:
Good catch! Still getting used to the way we're approaching the registration flow. Will add the token real quick!